cassandra-protocol/src/types/data_serialization_types.rs

```diff
@@ -151,6 +151,21 @@ pub fn decode_list(bytes: &[u8], version: Version) -> Result<Vec<CBytes>, io::Er
     Ok(list)
 }
 
+pub fn decode_vector(
```

Comment:
This should be explicitly named as decoding floats, since a Vector can have any internal type.